File: cmstestsuite/code/half-correct-fileio.rs
use std::fs::File;
use std::io::BufReader;
use std::io::prelude::*;

fn main()
{
    let mut input = BufReader::new(File::open("input.txt").unwrap());
    let mut output = File::create("output.txt").unwrap();
    let mut s = String::new();
    match input.read_line(&mut s)
    {
        Ok(_) =>
        {
            let n = s.trim().parse::<i32>().unwrap();
            let _ = write!(output, "correct {}", if n % 2 == 0 {n} else {0});
        },
        Err(why) => panic!("{}", why),
    };
}

File: cmstestsuite/code/half-correct-fileio.cs
using System;
using System.IO;

public static class Batchfileio
{
    public static void Main()
    {
        StreamReader reader = new StreamReader("input.txt");
        int n = int.Parse(reader.ReadLine());
        StreamWriter writer = new StreamWriter("output.txt");
        writer.WriteLine("correct " + (n % 2 != 0 ? n : 0));
        reader.Close();
        writer.Close();
    }
}
